cmd/boost/init.go

```diff
@@ -55,156 +63,576 @@ var initCmd = &cli.Command{
 	},
 	Before: before,
 	Action: func(cctx *cli.Context) error {
-		log.Info("Initializing boost repo")
-
-		ctx := lcli.ReqContext(cctx)
+		bp, err := initBoost(cctx)
+		if err != nil {
+			return err
+		}
 
-		log.Debug("Trying to connect to full node RPC")
+		lr, err := bp.repo.Lock(node.Boost)
+		if err != nil {
+			return err
+		}
+		defer lr.Close()
 
-		walletPSD, err := address.NewFromString(cctx.String("wallet-publish-storage-deals"))
+		ds, err := lr.Datastore(context.Background(), metadataNamespace)
 		if err != nil {
-			return fmt.Errorf("failed to parse wallet-publish-storage-deals: %s; err: %w", cctx.String("wallet-publish-storage-deals"), err)
+			return err
 		}
 
-		walletCP, err := address.NewFromString(cctx.String("wallet-collateral-pledge"))
+		log.Debug("Creating boost config")
+		var cerr error
+		err = lr.SetConfig(func(raw interface{}) {
+			rcfg, ok := raw.(*config.Boost)
+			if !ok {
+				cerr = xerrors.New("expected boost config")
+				return
+			}
+
+			asi, err := checkApiInfo(cctx.Context, cctx.String("api-sector-index"))
```

Comment:
Not sure if this context will be canceled on SIGTERM and all other signals. I think we should be using the ReqContext everywhere, as it is terminated on these signals.